### ISSUE TITLE
Elec. levels of ionized species not parsed correctly when reading reaction formula

### ISSUE THREAD

**Issue by jbscoggi** [mutationpp/Mutationpp#91]
I'm working on an Ar STS mechanism which has electron/atom impact ionization reactions where the ion Ar+ is split into two separate levels.  When reading these reactions, for example

```xml
<reaction formula="Ar(0)+e-=Ar+(0)+e-+e-">
```

I gen an error: 

```
M++ error: error parsing file.
file: Kapper_mech.xml
line: 3168
Species Ar is not in the mixture list!
```

Obviously, the problem seems to be that the parser sees `Ar+` and then thinks that is the species name without continuing to capture the `(0)` part.  

@amorosej, I think I remember you saying you had already fixed this in the past but I don't find your PR for this.  Could you please recall what the fix was?  Thanks.

**Comment by jbscoggi** [comment]
Never mind @amorosej, I think I have fixed this issue.  Waiting for the tests to complete to be sure.

**Comment by jbscoggi** [comment]
Fixed with #92, closing.
